features/cellular/framework/AT/AT_CellularContext.cpp

```diff
@@ -79,6 +81,25 @@ void AT_CellularContext::set_file_handle(FileHandle *fh)
     _at.set_file_handle(_fh);
 }
 
+#if (DEVICE_SERIAL && DEVICE_INTERRUPTIN && NSAPI_PPP_AVAILABLE)
+void AT_CellularContext::set_file_handle(UARTSerial *serial, PinName dcd_pin, bool active_high)
+{
+    tr_info("CellularContext serial %p", serial);
+    _dcd_pin = dcd_pin;
+    _active_high = active_high;
+    _fh = serial;
+    _at.set_file_handle(_fh);
+    enable_hup(false);
+}
+
+void AT_CellularContext::enable_hup(bool enable)
+{
+    if (_dcd_pin != NC) {
+        (static_cast<UARTSerial *>(_fh))->set_data_carrier_detect(enable ? _dcd_pin : NC, _active_high);
```

Comment:
Don't need the extra `()` - the `static_cast` has its own built-in which seem clear enough to me.

Reply:
fixed.